stl/src/special_math.cpp

```diff
@@ -23,16 +23,12 @@
 #include <boost/math/special_functions/bessel.hpp>
 #include <boost/math/special_functions/beta.hpp>
 #include <boost/math/special_functions/ellint_1.hpp>
-#include <boost/math/special_functions/ellint_2.hpp>
 #include <boost/math/special_functions/ellint_3.hpp>
 #include <boost/math/special_functions/expint.hpp>
 #include <boost/math/special_functions/hermite.hpp>
 #include <boost/math/special_functions/laguerre.hpp>
-#include <boost/math/special_functions/legendre.hpp>
 #include <boost/math/special_functions/spherical_harmonic.hpp>
-#include <boost/math/special_functions/zeta.hpp>
 #include <boost/math/tools/config.hpp>
-#include <boost/math/tools/precision.hpp>

```

Comment:
Each of the four headers being struck here is the documented source of a component used directly in this file. Why are we removing this headers?

Reply:
Perhaps the documentation is out of date or simply wrong.

Reply:
I suspect that the situation with Boost is very similar to the Standard Library itself - header `<cats.hpp>` is documented to provide `meow()`, but as an implementation detail, `<giraffes.hpp>` happens to drag in `meow()`. Implementation details can change, especially for external components, so we shouldn't rely on them.

Therefore I believe that such changes should be reverted - only if the documentation indicates that some headers being included are truly not necessary, should they be removed. I do recall auditing this specific file years ago, although things may have changed since then, so I don't claim that analysis is still up to date.